torch/_export/__init__.py

```diff
@@ -145,12 +143,18 @@ def export(
         constraints: A optional list of constraints on the dynamic arguments specifying
             their possible range of their shapes
 
+        decompositions (Dict): A dictionary to define the decomposition of
+            larger Aten ops into simpler or core Aten ops.
+
     Returns:
         An ExportedProgram containing the traced method.
     """
     constraints = constraints or []
     kwargs = kwargs or {}
 
+    if decomposition_table is None:
+        decomposition_table = core_aten_decompositions()
```

Comment:
Do we want to keep this only for AOTInductor? Also when decomposition_table is not None, you want to merge the two dicts.

Reply:
`dict.update` can take precedence, but I guess arguably either way is fine. Just need to make the behavior clear in the user API comment/doc.

Reply:
Hmm, but what if a user removed something from the core aten decomp table? Then by adding it, it'll get decomposed again.

Reply:
I guess from API design point of view you can do two things as Bin suggested:
1. overwrite
2. merge
overwrite is fine but the downside is user is implicitly depending on the fact that export() uses core aten IR by default, which is a little bit leaky abstraction to me, but I guess in practice it's fine?

If we wanna go with merge route, then we need an "addition set" and a "minus set" which are two arguments.

Reply:
My preference is to go with overwriting, but another idea is, what if we use a context manager to specify the decomp table rather having it as an argument to export? 

My feeling is that the manipulation of the decomp table will be used by super users, so if they do want to specify their own decompositions then they can specify it through a context manager like:
```
with set_export_decomp_table(inductor_decomp):
    ep = export(...)
```

Reply:
I think it's still leaky whether you make it context manager or not. User just express their intention differently here:
1. For overwrite, user basically says: no matter what export() does, use MY decomp table.
2. For merge, use says: on top of export() decomp behavior, I want these to be decomped and those to be not decomped. 

I think 1 and 2 are not the same although in practice they could achieve the same thing. I think context manager v.s. argument is just implicit vs explicit over an API which is orthogonal to the semantics difference I mentioned above.

Reply:
Chatted offline: since we don't have a strong use case for passing in the decomp table yet, we will just patch the decomp table directly in our helper function